src/platform_impl/apple/appkit/window_delegate.rs

```diff
@@ -1837,6 +1846,32 @@ impl WindowExtMacOS for WindowDelegate {
     fn is_borderless_game(&self) -> bool {
         self.ivars().is_borderless_game.get()
     }
+
+    fn set_unified_titlebar(&self, unified_titlebar: bool) {
+        let window = self.window();
+
+        if unified_titlebar {
+            let mtm = MainThreadMarker::from(self);
+
+            unsafe {
+                window.setToolbar(Some(&NSToolbar::new(mtm)));
```

Comment:
Is it really necessary for us to add a toolbar for this to work? What happens if no toolbar is set? Could you document the behaviour as a code comment?

Reply:
Yeah I can add a comment, not at this exact moment, but I will when I get the chance later today